File: SDKs/netboxdemo.com/2.4/typescript/src/sdk/models/operations/dcimdevicebayslist.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class DcimDeviceBaysListQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=device" })
  device?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=device_id" })
  deviceId?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @Metadata({ data: "queryParam, style=form;explode=true;name=name" })
  name?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=offset" })
  offset?: number;

  @Metadata({ data: "queryParam, style=form;explode=true;name=tag" })
  tag?: string;
}


export class DcimDeviceBaysListRequest extends SpeakeasyBase {
  @Metadata()
  queryParams: DcimDeviceBaysListQueryParams;
}


export class DcimDeviceBaysList200ApplicationJson extends SpeakeasyBase {
  @Metadata({ data: "json, name=count" })
  count: number;

  @Metadata({ data: "json, name=next" })
  next?: string;

  @Metadata({ data: "json, name=previous" })
  previous?: string;

  @Metadata({ data: "json, name=results", elemType: shared.DeviceBay })
  results: shared.DeviceBay[];
}


export class DcimDeviceBaysListResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  dcimDeviceBaysList200ApplicationJsonObject?: DcimDeviceBaysList200ApplicationJson;
}

File: SDKs/netboxdemo.com/2.4/go/pkg/models/operations/dcimdevicebayslist.go
package operations

import (
	"openapi/pkg/models/shared"
)

type DcimDeviceBaysListQueryParams struct {
	Device   *string `queryParam:"style=form,explode=true,name=device"`
	DeviceID *string `queryParam:"style=form,explode=true,name=device_id"`
	Limit    *int64  `queryParam:"style=form,explode=true,name=limit"`
	Name     *string `queryParam:"style=form,explode=true,name=name"`
	Offset   *int64  `queryParam:"style=form,explode=true,name=offset"`
	Tag      *string `queryParam:"style=form,explode=true,name=tag"`
}

type DcimDeviceBaysList200ApplicationJSON struct {
	Count    int64              `json:"count"`
	Next     *string            `json:"next,omitempty"`
	Previous *string            `json:"previous,omitempty"`
	Results  []shared.DeviceBay `json:"results"`
}

type DcimDeviceBaysListRequest struct {
	QueryParams DcimDeviceBaysListQueryParams
}

type DcimDeviceBaysListResponse struct {
	ContentType                                string
	StatusCode                                 int64
	DcimDeviceBaysList200ApplicationJSONObject *DcimDeviceBaysList200ApplicationJSON
}
